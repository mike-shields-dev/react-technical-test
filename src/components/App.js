import React, { useState } from "react";
import "../styles/app.css";
import Search from "./Search";
import SearchResults from "./SearchResults";
import logo from "../assets/img/NASA_Worm_logo.svg";

const App = () => {
  const [results, setResults] = useState([]);
  return (
    <div className="app">
      <header>
        <img className="nasa-logo" src={logo} alt="nasa logo" />
        <Search setSearchResults={setResults} />
      </header>
      <SearchResults results={results} />
    </div>
  );
};

export default App;
