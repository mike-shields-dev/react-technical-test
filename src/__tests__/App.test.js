import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the NASA logo", () => {
    render(<App />);

    expect(screen.getByAltText("nasa logo")).toBeInTheDocument();
  });

  it("renders Search component", () => {
    render(<App />);

    expect(screen.getByRole("form")).toBeInTheDocument();
  });
});
