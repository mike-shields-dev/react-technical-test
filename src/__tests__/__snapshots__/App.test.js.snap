// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <header>
      <img
        alt="nasa logo"
        class="nasa-logo"
        src="NASA_Worm_logo.svg"
      />
      <form
        class="search-form"
        name="search-form"
      >
        <input
          class="search-form__input"
          placeholder="enter your search"
          type="text"
          value=""
        />
        <button
          class="search-form__button"
          type="submit"
        >
          Go!
        </button>
      </form>
    </header>
    <p
      class="no-results"
    >
      No results
    </p>
  </div>
</DocumentFragment>
`;
